nimbella/nimbella.py

```diff
@@ -49,13 +48,14 @@ def storage(web=False):
         raise Exception('Object store credentials are not available.')
     try:
         creds = json.loads(creds)
-        client = gstorage.Client(
-            creds['project_id'], credentials=service_account.Credentials.from_service_account_info(creds))
     except:
         raise Exception(
             'Insufficient information in provided credentials or credentials were invalid.')
 
-    hostpart = '-'.join(apiHost.replace('https://', '').split('.'))
-    datapart = '' if web else 'data-'
-    bucket = datapart + namespace + '-' + hostpart
-    return client.get_bucket(bucket)
+    provider_id = creds.get('provider', '@nimbella/storage-gcs')
```

Comment:
The `creds.get` call can return `None` because customers who got their storage credentials before we had storage abstraction don't have a provider field.   This needs to default to the GCS provider.   Also noted elsewhere.

Reply:
The second argument to the `creds.get` method (`@nimbella/storage-gcs`) is a default value to be returned in case the Python dict does not have a property. This should handle the default case unless I've missed something?

Reply:
Ooops.   You are right.   Should have noticed that.   Sorry.